RecoBTag/FeatureTools/interface/NeutralCandidateConverter.h

```diff
@@ -32,18 +32,20 @@ namespace btagbtvdeep {
                                         NeutralCandidateFeatures& n_pf_features) {
     const auto* patJet = dynamic_cast<const pat::Jet*>(&jet);
 
-    if (!patJet) {
-      throw edm::Exception(edm::errors::InvalidReference) << "Input is not a pat::Jet.";
-    }
     // Do Subjets
-    if (patJet->nSubjetCollections() > 0) {
-      auto subjets = patJet->subjets();
-      // sort by pt
-      std::sort(subjets.begin(), subjets.end(), [](const edm::Ptr<pat::Jet>& p1, const edm::Ptr<pat::Jet>& p2) {
-        return p1->pt() > p2->pt();
-      });
-      n_pf_features.drsubjet1 = !subjets.empty() ? reco::deltaR(*n_pf, *subjets.at(0)) : -1;
-      n_pf_features.drsubjet2 = subjets.size() > 1 ? reco::deltaR(*n_pf, *subjets.at(1)) : -1;
+    if (patJet) {
```

Comment:
why all this code duplication?

Reply:
Could you please clarify the comment about what code exactly is duplicated in this file?

Reply:
the code in this file looks identical to the one in the previous one.

Reply:
The first file is about Charged candidates and this one is about the Neutral ones. Some elements look similar indeed but they define 2 different inputs used by the neural network.